criu/mount.c

```diff
@@ -2942,6 +2944,68 @@ static int get_mp_mountpoint(char *mountpoint, struct mount_info *mi, char *root
 	return 0;
 }
 
+static char *mount_update_lsm_context(char *mount_opts)
+{
+	cleanup_free char *before_context = NULL;
+	cleanup_free char *new_options = NULL;
+	char *other_options;
+	char *context_start;
+	char *context_end;
+	char *old_context;
+	int ret;
+
+	old_context = strstr(mount_opts, CONTEXT_OPT);
+
+	if (!old_context || !opts.lsm_mount_context)
+		return xstrdup(mount_opts);
+
+	/*
+	 * If the user specified a different mount_context we need
+	 * to replace the existing mount context in the mount
+	 * options with the one specified by the user.
+	 *
+	 * The original mount options will be something like:
+	 *
+	 *  context="system_u:object_r:container_file_t:s0:c82,c137",inode64
+	 *
+	 * and it needs to be replaced with opts.lsm_mount_context.
+	 *
+	 * The content between 'context=' and ',inode64' will be replaced
+	 * with opts.lsm_mount_context in quotes.
+	 */
+
+	/* Skip 'context=' */
+	context_start = old_context + strlen(CONTEXT_OPT);
+	if (context_start[0] == '"' && context_start + 1 < mount_opts + strlen(mount_opts)) {
+		/* Skip quotes */
+		context_end = strchr(context_start + 1, '"');
+		if (!context_end) {
+			pr_err("Failed parsing mount option 'context'\n");
+			return NULL;
+		}
+	} else {
+		context_end = context_start;
+	}
+
+	/* Find next after optionally skipping quotes. */
+	other_options = strchr(context_end, ',');
+
+	before_context = xstrdup(mount_opts);
+	pr_debug("%p\n", before_context);
+	if (unlikely(!before_context))
+		return NULL;
+	before_context[context_start - mount_opts] = 0;
+
+	ret = asprintf(&new_options, "%s\"%s\"%s", before_context,
+			opts.lsm_mount_context, other_options ? other_options : "");
+	pr_debug("ret %d\n", ret);
```

Comment:
The information printed with this `pr_debug` doesn't look useful.

Reply:
Yes, that is wrong.

Reply:
removed the pr_debug